src/gui/wizard/owncloudadvancedsetuppage.cpp

```diff
@@ -158,6 +175,50 @@ void OwncloudAdvancedSetupPage::initializePage()
     _ui.confCheckBoxSize->setChecked(newFolderLimit.first);
     _ui.confSpinBox->setValue(newFolderLimit.second);
     _ui.confCheckBoxExternal->setChecked(cfgFile.confirmExternalStorage());
+
+    // Reset user avatar
+    const auto appIcon = Theme::instance()->applicationIcon();
+    _ui.lServerIcon->setPixmap(appIcon.pixmap(48));
+    // Fetch user avatar
+    auto account = _ocWizard->account();
+    auto avatarJob = new AvatarJob(account, account->davUser(), 64, this);
+    avatarJob->setTimeout(20 * 1000);
+    QObject::connect(avatarJob, &AvatarJob::avatarPixmap, this, [this](const QImage &avatarImage) {
+        if (avatarImage.isNull()) {
+            return;
+        }
+        auto avatarPixmap = QPixmap::fromImage(AvatarJob::makeCircularAvatar(avatarImage));
+        _ui.lServerIcon->setPixmap(avatarPixmap);
+    });
+    avatarJob->start();
+
+    // Fetch user data
+    auto userJob = new JsonApiJob(account, QLatin1String("ocs/v1.php/cloud/user"), this);
+    userJob->setTimeout(20 * 1000);
+    connect(userJob, &JsonApiJob::jsonReceived, this, [this](const QJsonDocument &json) {
+        auto objData = json.object().value("ocs").toObject().value("data").toObject();
+        const auto displayName = objData.value("display-name").toString();
+        _ui.userNameLabel->setText(displayName);
+    });
+    userJob->start();
+
+    const auto serverUrl = account->url().toString();
+    setServerAddressLabelUrl(serverUrl);
+    const auto userName = account->davDisplayName();
+    _ui.userNameLabel->setText(userName);
+}
+
+void OwncloudAdvancedSetupPage::setServerAddressLabelUrl(const QString &url)
+{
+    auto prettyUrl = url;
+
+    const QString httpsPrefix("https://");
```

Comment:
I guess this logic is too verbose. What I suggest is you could use **QUrl** as the argument type of **setServerAddressLabelUrl**. Then, you can just use `QUrl::scheme` to get either **http** or **https**. Then you can call `_ui.serverAddressLabel->setText` as following `const auto scheme = url.scheme(); const QString& prettyUrl = url.toString().replace(url.indexOf(scheme), scheme.size(), ""); _ui.serverAddressLabel->setText(prettyUrl)`.

You will benefit from always having the proper prefix. Right now, something bad may happen if **url.indexOf(httpPrefix)** returns **-1**. In fact, you are trying to replace both "http" and "https" here. Which concerns me.

Another benefit, is, you won't need to cast **QUrl** to QString in place of **setServerAddressLabelUrl** call.

Reply:
I like the QUrl approach. However, in my tests it was safe to let `url.indexOf()`return -1. The idea was: First search for `https://` and replace it. The replace statement with `http://` then does not have any effect. Otherwise if the replace call with `https://` does nothing then the replace call with `http://` takes effect.

Reply:
I like @allexzander proposal to use QUrl. There's always a danger lurking when we try to do partial URL parsing by hand while QUrl is a fine reusable parser.

That being said, since we get more guarantees out of QUrl `toString()` call I think there's slightly simpler than his proposed replace:
`const auto prettyUrl = url.toString().mid(scheme.size() + 3); // + 3 because we need to remove :// as well`

Also if you got a QUrl you can check this is actually a valid one before proceeding, while with QString you assume it is and hope for the best. So going with QUrl this could be all guarded with:
```
if (!url.isValid()) {
    return;
}
```